src/commands/lambda/__tests__/flare.test.ts

```diff
@@ -1,21 +1,444 @@
+      fs.writeFileSync = jest.fn().mockImplementation(() => {
+        throw new Error('MOCK ERROR: Unable to save zip')
+      })
+
+      await expect(zipContents(MOCK_FILE_PATH, MOCK_ZIP_PATH)).rejects.toMatchSnapshot()
+
+      expect(fs.readFileSync).toHaveBeenCalledWith(MOCK_FILE_PATH, 'utf8')
+      expect(mockJSZip.file).toHaveBeenCalledWith(MOCK_FILE_PATH, expect.anything())
+      expect(mockJSZip.generateAsync).toHaveBeenCalledWith({type: 'nodebuffer'})
+      expect(fs.writeFileSync).toHaveBeenCalled()
+    })
   })
 
-  it('prints dry-run header', async () => {
-    const cli = makeCli()
-    const context = createMockContext()
-    const code = await cli.run(['lambda', 'flare', '-d'], context as any)
-    const output = context.stdout.toString()
-    expect(code).toBe(0)
-    expect(output).toMatchSnapshot()
+  describe('send to Datadog', () => {
+    // File system mocks
+    beforeAll(() => {
+      fs.writeFileSync = jest.fn().mockImplementation(() => {})
+      fs.mkdirSync = jest.fn().mockImplementation(() => {})
+    })
+    process.env = {['DATADOG_API_KEY']: MOCK_API_KEY}
+
+    it('successfully adds zip file to FormData', async () => {
+      const appendSpy = jest.spyOn(FormData.prototype, 'append')
+      const cli = makeCli()
+      const context = createMockContext()
+      await cli.run(VALID_INPUT, context as any)
+      expect(appendSpy).toHaveBeenCalled()
+      appendSpy.mockRestore()
+      const output = context.stdout.toString()
+      expect(output).toMatchSnapshot()
+    })
+
+    it('successfully sends request to Datadog', async () => {
+      const postSpy = jest.spyOn(axios, 'post').mockResolvedValue({status: 200})
+      const cli = makeCli()
+      const context = createMockContext()
+      await cli.run(VALID_INPUT, context as any)
+      expect(postSpy).toHaveBeenCalledWith(
+        expect.any(String),
+        expect.any(FormData),
+        expect.objectContaining({
+          headers: expect.objectContaining({
+            'DD-API-KEY': MOCK_API_KEY,
+          }),
+        })
+      )
+      const output = context.stdout.toString()
+      expect(output).toMatchSnapshot()
+      postSpy.mockRestore()
+    })
+
+    it('does not send request to Datadog when a dry run', async () => {
+      const postSpy = (axios.post = jest.fn().mockRejectedValue({status: 500}))
+      const cli = makeCli()
+      const context = createMockContext()
+      const code = await cli.run([...VALID_INPUT, '-d'], context as any)
+      expect(code).toBe(0)
+      expect(postSpy).not.toHaveBeenCalled()
+      const output = context.stdout.toString()
+      expect(output).toMatchSnapshot()
+      postSpy.mockRestore()
+    })
+  })
+
+  describe('AWS Lambda configuration', () => {
+    it('stops and prints error when getLambdaFunctionConfig fails', async () => {
+      require('../functions/commons').getLambdaFunctionConfig.mockImplementation(() => {
+        throw new Error('MOCK ERROR: Some API error')
+      })
```

Comment:
Let's not do `require()` every time we want to mock something.

On the other hand, require it at the top, and then do:

```js
;(functionToMock as any).mockImplementation(() => Promise.reject('Unexpected error'))
```

Or whatever the outcome you are trying to achieve.